app/src/main/java/group1/cpsc319/plurilock_client/DataCollectionUtils/SocketClient.java

```diff
@@ -84,6 +84,8 @@ public void onOpen(ServerHandshake serverHandshake) {
                 Log.i("Websocket", "Opened");
                 mWebSocketClient.send("Hello from Plurilock!");
 
+                while (activity == null) {}
```

Comment:
This is a terrible, terrible thing that I've done. I apologize.

For more context, when GestureActivity gets the instance of DataManager, DataManager gets an instance of SocketClient, using the SocketClient.getInstance() method. This doesn't pass in an activity that's needed to create toasts. While this code can be run and the app won't crash, it's nicer to have consistent behaviour. Adding the activity to SocketClient is done very soon after GestureActivity is created, so there's no perceptible lag.